hazelcast/src/test/java/com/hazelcast/map/listener/EntryUpdatedListenerTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.listener;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.map.MapInterceptor;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class EntryUpdatedListenerTest extends HazelcastTestSupport {
```

Comment:
I think it would be good to parametrize this test and run it with `OBJECT` & `BINARY` in-memory formats (like we do in `EntryProcessorTest`).

Reply:
> I think it would be good to parametrize this test and run it with `OBJECT` & `BINARY` in-memory formats (like we do in `EntryProcessorTest`).

For `OBJECT` serialization, the the `EntryUpdatedListener`'s `EntryEvent.getOldValue()` returns `null` (even in `master`), whereas for `BINARY`, it's populated.

This behaviour is specified in `EntryOperator.getOrNullOldValue()` and I *think* it's done this way to try to prevent similar mutation side-effects.

For both, the `MapInterceptor.interceptPut`'s `oldValue` is populated correctly.

Really this test should be inside of the `EntryProcessorTest` class, but because of this inconsistency it can only be run for `BINARY` serialization and so can't go in the parameterized test class.